web/rootfs/defaults/settings-config.js

```diff
@@ -135,17 +136,17 @@ config.etherpad_base = '{{ $PUBLIC_URL }}/etherpad/p/';
 // Recording.
 //
 
-{{ if $ENABLE_RECORDING -}}
+{{ if or $ENABLE_RECORDING .Env.DROPBOX_APPKEY $ENABLE_LIVESTREAMING  -}}
```

Comment:
I don't think we should check for the Dropbox app key here. ENABLE_RECORDING should also be set, if not we should ignore that.

Reply:
Hey @saghul ,

Thanks for the feedback!

What I'm looking for is a way to enable Dropbox but disabling the recordingService (because in the new version, both services can be enabled at the same time).

So what I want in the end is to have a file like this:

```js
// Whether to enable file recording or not
config.recordingService.enabled = false;
config.liveStreamingEnabled = true;

// Enable the dropbox integration.
if (!config.hasOwnProperty('dropbox')) config.dropbox = {};
config.dropbox.appKey = 'my_dropbox_key';
```

To do this, I though I would use those environment variables:

```
ENABLE_RECORDING=false
DROPBOX_APPKEY=my_dropbox_key
```

It slightly modifies the semantics of `ENABLE_RECORDING` because now, `ENABLE_RECORDING` refers only to the recordingService (but this is coherent with the change in the Jibri configuration).



If you want `ENABLE_RECORDING` to be a generic environment variable spawning both the "recording service" and dropbox, maybe I can find another environment variable that specifically disables the recording service?

Something like a new "DISABLE_SERVICE_RECORDING"?

```
{{ if $ENABLE_RECORDING -}}

config.hiddenDomain = '{{ $XMPP_RECORDER_DOMAIN }}';

if (!config.hasOwnProperty('recordingService')) config.recordingService = {};

// Whether to enable file recording or not
config.recordingService.enabled = {{ not $DISABLE_SERVICE_RECORDING }};

// Whether to enable live streaming or not.
config.liveStreamingEnabled = {{ $ENABLE_LIVESTREAMING }};

{{ if .Env.DROPBOX_APPKEY -}}
// Enable the dropbox integration.
if (!config.hasOwnProperty('dropbox')) config.dropbox = {};
config.dropbox.appKey = '{{ .Env.DROPBOX_APPKEY }}';
{{ if .Env.DROPBOX_REDIRECT_URI -}}
// A URL to redirect the user to, after authenticating
// by default uses:
// 'https://jitsi-meet.example.com/static/oauth.html'
config.dropbox.redirectURI = '{{ .Env.DROPBOX_REDIRECT_URI }}';
{{ end -}}
{{ end -}}
```

In this case, to achieve a "dropbox-only" recording, one would have to use:

```
ENABLE_RECORDING=true
DISABLE_SERVICE_RECORDING=true
DROPBOX_APPKEY=my_dropbox_key
```

What do you think works best?

Reply:
Let me double check, I think you're right.